neo/SmartContract/InteropService.NEO.cs

```diff
@@ -231,77 +229,56 @@ private static bool Contract_Create(ApplicationEngine engine)
             byte[] script = engine.CurrentContext.EvaluationStack.Pop().GetByteArray();
             if (script.Length > 1024 * 1024) return false;
 
-            var manifest = Encoding.UTF8.GetString(engine.CurrentContext.EvaluationStack.Pop().GetByteArray());
-            if (manifest.Length >= ContractManifest.MaxLength) return false;
+            var manifest = engine.CurrentContext.EvaluationStack.Pop().GetString();
+            if (manifest.Length > ContractManifest.MaxLength) return false;
 
             UInt160 hash = script.ToScriptHash();
             ContractState contract = engine.Snapshot.Contracts.TryGet(hash);
-            if (contract == null)
+            if (contract != null) return false;
+            contract = new ContractState
             {
-                contract = new ContractState
-                {
-                    Script = script,
-                    Manifest = ContractManifest.Parse(manifest)
-                };
-
-                if (!contract.Manifest.IsValid()) return false;
-
-                engine.Snapshot.Contracts.Add(hash, contract);
-            }
-            engine.CurrentContext.EvaluationStack.Push(StackItem.FromInterface(contract));
-            return true;
-        }
-
-        private static bool Contract_UpdateManifest(ApplicationEngine engine)
-        {
-            if (engine.Trigger != TriggerType.Application) return false;
-
-            var manifest = Encoding.UTF8.GetString(engine.CurrentContext.EvaluationStack.Pop().GetByteArray());
-            if (manifest.Length >= ContractManifest.MaxLength) return false;
-
-            var contract = engine.Snapshot.Contracts.TryGet(engine.CurrentScriptHash);
-            if (contract == null) return false;
-
-            contract.Manifest = ContractManifest.Parse(manifest);
+                Script = script,
+                Manifest = ContractManifest.Parse(manifest)
+            };
 
             if (!contract.Manifest.IsValid()) return false;
-            if (contract.Manifest.Hash != engine.CurrentScriptHash) return false;
-            if (contract.Manifest.Abi.Hash != engine.CurrentScriptHash) return false;
 
-            engine.Snapshot.Contracts.Add(contract.ScriptHash, contract);
+            engine.Snapshot.Contracts.Add(hash, contract);
             engine.CurrentContext.EvaluationStack.Push(StackItem.FromInterface(contract));
             return true;
         }
 
-        private static bool Contract_Migrate(ApplicationEngine engine)
+        private static bool Contract_Update(ApplicationEngine engine)
         {
             if (engine.Trigger != TriggerType.Application) return false;
```

Comment:
We will use the same Method for migrate and update manifest?

Reply:
Yes. `Update(script, null)` for updating script; `Update(null, manifest)` for updating manifest; `Update(script, manifest)` for updating both.